cms/djangoapps/contentstore/views/course.py

```diff
@@ -283,6 +294,28 @@ def course_rerun_handler(request, course_key_string):
             })
 
 
+@login_required
+@ensure_csrf_cookie
+@require_GET
```

Comment:
All new APIs should be built using Django REST Framework so I think this one should be too. It would also be good to support OAuth as we won't always want to use session authentication to kick off indexing.

See this best practices guide for details:
https://openedx.atlassian.net/wiki/display/AC/API+Thoughts